go.mod refers to Tencent Cloud SDK versions that have been deleted upstream

### Community note

Trying to build opentofu from source (either the 1.10.3 tarball or a git checkout) results in

```
go: downloading github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/sts v1.0.588
go: downloading github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/tag v1.0.233
# github.com/opentofu/opentofu/cmd/tofu
internal/backend/remote-state/cos/backend.go:23:2: reading github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/sts/go.mod at revision tencentcloud/sts/v1.0.588: unknown revision tencentcloud/sts/v1.0.588
# github.com/opentofu/opentofu/cmd/tofu
internal/backend/remote-state/cos/backend.go:24:2: reading github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/tag/go.mod at revision tencentcloud/tag/v1.0.233: unknown revision tencentcloud/tag/v1.0.233
FAIL    github.com/opentofu/opentofu/cmd/tofu [setup failed]
```

Sure enough, the referenced tags don't exist in https://github.com/TencentCloud/tencentcloud-sdk-go -- there is a `tencentcloud/common/v1.0.588`, but not a `tencentcloud/sts/v1.0.588`. The closest matches are `tencentcloud/sts/v1.0.586` and `tencentcloud/sts/v1.0.591`, the current version seems to be `tencentcloud/sts/v1.0.1200`.
Similary., `tencentcloud/tag/v1.0.233` doesn't exist, but `tencentcloud/common/v1.0.233`, `tencentcloud/tag/v1.0.232`, `tencentcloud/tag/v1.0.279` and `tencentcloud/tag/v1.0.1200` do.

Chances are tencentcloud removed old tags from its repositories.


Editing go.mod to use v1.0.1200 for tencentcloud/sts and tencentcloud/tag (and adjusting go.sum accordingly) results in a successful build here. (Can't check if it actually works with tencentcloud though).

### OpenTofu Version

```shell
1.10.3, and it looks the same on master branch
```

### OpenTofu Configuration Files

n/a

### Debug Output

n/a

### Expected Behavior

It compiles

### Actual Behavior

Doesn't compile, errors out when trying to download the nonexistent tencentcloud libraries

### Steps to Reproduce

`make`

or

`go mod download`

### Additional Context

_No response_

### References

_No response_

--- a/go.mod
+++ b/go.mod
@@ -84,8 +84,8 @@ require (
 	github.com/posener/complete v1.2.3
 	github.com/spf13/afero v1.9.3
 	github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common v1.0.588
-	github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/sts v1.0.588
-	github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/tag v1.0.233
+	github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/sts v1.0.587-0.20230206000712-97469a3dcd4e
+	github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/tag v1.0.233-0.20210823002710-8078545fa058
 	github.com/tencentyun/cos-go-sdk-v5 v0.7.29
 	github.com/tombuildsstuff/giovanni v0.15.1
 	github.com/xanzy/ssh-agent v0.3.1

